protocol/contracts/governance/ElderElection.sol

```diff
@@ -0,0 +1,177 @@
+
+
+    constructor(
+        ERC721 _templars,
+        IERC20 _nominationFeeToken,
+        uint256 _nominationFee
+    ) EIP712("ElderElection", "1") {
+        templars = _templars;
+        nominationFeeToken = _nominationFeeToken;
+        nominationFee = _nominationFee;
+    }
+
+    /**
+     * @notice Change the nomination fee
+     * 
+     * Designed to be low, but a defense in depth to stop peeps putting up a nomination
+     * for the lolz.
+     */
+    function setNominationFee(uint256 _nominationFee) external onlyOwner {
+        nominationFee = _nominationFee;
+    }
+
+    /**
+     * @notice Update the merkle tree root for temple roles.
+     */
+    function setTemplarRolesMerklRoot(bytes32 _templarRolesMerkleRoot) external onlyOwner {
+        templarRolesMerkleRoot = _templarRolesMerkleRoot;
+    }
+
+    /**
+     * @notice Nominate the given discord ID a candidate.
+     * @dev The given address must hold the NFT for the specified discord id
+     */
+    function nominate(
+        address account,
+        uint256 discordId
+    ) external {
+        if (templars.ownerOf(discordId) != account) {
+            revert NotFromTemplar(account, discordId);
+        }
+
+        if (!candidates[discordId]) {
+
+          if (nominationFee > 0) {
+              nominationFeeToken.safeTransferFrom(account, address(this), nominationFee);
+          }
+          candidates[discordId] = true;
+          emit UpdateNomination(discordId, candidates[discordId]);
+        }
+    }
+
+    /**
+     * @notice Remove the given discord ID as a candidate
+     * @dev The given address must hold the NFT for the specified discord id
+     */
+    function resign(
+        address account,
+        uint256 discordId
+    ) external {
+        if (templars.ownerOf(discordId) != account) {
+            revert NotFromTemplar(account, discordId);
+        }
+
+        if (candidates[discordId]) {
+          candidates[discordId] = false;
+          emit UpdateNomination(discordId, candidates[discordId]);
+        }
+    }
+
+    function setEndorsements(uint256[] memory discordIds) external {
+        _setEndorsements(msg.sender, discordIds);
+    }
+
+    function _setEndorsements(address account, uint256[] memory discordIds) internal {
+        endorsementsBy[account] = discordIds;
+        emit UpdateEndorsements(account, discordIds);
+    }
+
+    /**
+     * @notice Toggle the 
+     * (assuming the owner has given authority for the caller to act on their behalf)
+     *
+     * @dev amount is explicit, to allow use case of partial vault withdrawals
+     */
+    // TODO(butlerji): Need to test this works. The idea is I'd sign a message on mainnet, which gets sent to our backend, then sent to 
+    //                 the mempool on whatever L2 we are running this on.
+    function relayedSetEndorsementsFor(address account, uint256[] memory discordIds, uint256 deadline, bytes calldata signature) public {
```

Comment:
Can be external, no need for private here?

Reply:
which "private" ?

Reply:
sorry - i meant 'public', updated